backend/cybnetics/model_builder.py

```diff
@@ -0,0 +1,110 @@
+# pooling is not set if the layer doesn't use pooling
+def make_model_class(layers, pools):
+    for pool in pools:
+        try:
+            class_name = pool['type']
+            args = pool['args']
+            kwargs = pool.get('kwargs', {})
+        except:
+            raise BadModelSpec('pool should specify type and args')
+        try:
+            pool_class = getattr(nn, class_name)
+        except:
+            raise BadModelSpec(f'pool type {class_name} was not found')
+        pool['instance'] = pool_class(*args, **kwargs)
+
+    for layer in layers:
+        # get params from layer dict
+        try:
+            class_name = layer['type']
+            args = layer['args']
+            if class_name != 'view':
+                name = layer['name']
+            activation = layer.get('activation')
+            kwargs = layer.get('kwargs', {})
+        except:
+            raise BadModelSpec('layer should specify type args, and activation')
+
+        # view special case/hack
+        if class_name == 'view':
+            layer['instance'] = ViewAdapter(*args, **kwargs)
+            layer['pool'] = identity
+            layer['activation'] = identity
+            continue
+
+        # initialize instance, pool and activation
+        try:
+            layer_class = getattr(nn, class_name)
+        except:
+            raise BadModelSpec(f'layer type {class_name} was not found')
+        try:
+            if not activation:
+                # another hack for the last layer :/
+                layer['activation'] = identity
+            else:
+                layer['activation'] = getattr(F, activation)
+        except:
+            raise BadModelSpec(f'couldn\'t find activation function {activation}')
+        pool = layer.get('pool')
+        if pool is not None:
+            if pool >= len(pools) or pool < 0:
+                raise BadModelSpec(f'invalid pool number {pool}')
+            layer['pool'] = pools[int(pool)]['instance']
+        else:
+            # if no pool use identity for simplicity
+            layer['pool'] = identity
+        layer['instance'] = layer_class(*args, **kwargs)
+        
+    class ModelClass(nn.Module):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.layers = layers
+            self.pools = pools
+            # set layers as attrs so loading the tate dict works
+            for layer in self.layers:
+                if layer['type'] == 'view':
+                    continue
+                if getattr(self, layer['name'], None):
+                    raise BadModelSpec(f'invalid layer name {layer["name"]}')
+                setattr(self, layer['name'], layer['instance'])
+            
+        def forward(self, x):
+            for layer in self.layers:
+                x = layer['pool'](layer['activation'](layer['instance'](x)))
+            return x
+    return ModelClass
+
+def make_cfair10_example():
```

Comment:
here is an example of input for the `layers` and `pools` parameters, there is also a new parameter to create called `color` which is true if the image is in color @dhruvilp

Reply:
@mjrb yeah plz update endpoint doc